src/client/pages/Section/DataTable/Table/hooks/useParsedTable.ts

```diff
@@ -0,0 +1,60 @@
+import { useMemo } from 'react'
+
+import { AssessmentName, Row, RowType, Table } from 'meta/assessment'
+import { RecordAssessmentData } from 'meta/data'
+
+import { useCycle } from 'client/store/assessment'
+import { useShowOriginalDatapoints } from 'client/store/ui/assessmentSection'
+import { useCountryIso } from 'client/hooks'
+
+import { parseTable } from './_parseTable'
+
+type Props = {
+  assessmentName: AssessmentName
+  data: RecordAssessmentData
+  table: Table
+}
+
+type Returned = {
+  headers: Array<string>
+  noticeMessages: Array<Row>
+  rowsData: Array<Row>
+  table: Table
+}
+
+export const useParsedTable = (props: Props): Returned => {
+  const { assessmentName, data, table: tableProps } = props
+
+  const countryIso = useCountryIso()
+  const cycle = useCycle()
+  const showODP = useShowOriginalDatapoints()
+
+  return useMemo<Returned>(() => {
+    const { headers, table } = parseTable({
+      assessmentName,
+      countryIso,
+      cycle,
+      data,
+      showODP,
+      table: tableProps,
+    })
+
+    const rowsData: Array<Row> = []
+    const noticeMessages: Array<Row> = []
+
+    table.rows.forEach((row) => {
+      if (row.props.type === RowType.noticeMessage) {
+        noticeMessages.push(row)
+      } else if (!row.props.hidden && row.props.type !== RowType.header) {
+        rowsData.push(row)
```

Comment:
Then we need to return the data rows count and notice messages count to be able to calculate the lastRow prop.

Currently it is:
```
lastRow={
  index === rowsData.length - 1 || (index === rowsData.length - 1 && !Objects.isEmpty(noticeMessages))
}
```
if we return all rows together then it would be something like:
```
const { dataRowsCount, headers, noticeMessagesCount, rows, table } = useParsedTable({ assessmentName, data, table: tableProps })
.
.
.

lastRow={
  index === rows.length - 1 || (index === dataRowsCount -1 && noticeMessagesCount > 0)
}
```
What do you think? we do that instead?